packages/blade/src/components/Button/Button/Button.tsx

```diff
@@ -35,6 +36,13 @@ type ButtonCommonProps = {
   isLoading?: boolean;
   accessibilityLabel?: string;
   type?: 'button' | 'reset' | 'submit';
+
+  /**
+   * It is exposed for internal usage with tooltip.
+   *
+   * @private
+   */
+  'aria-describedby'?: string;
```

Comment:
Shoudn't we make it `ariaDescribedBy`? would make more sense with rest of naming

Reply:
Can we do this: 

Remove this type-definition so users don't get the intellisence? 
In the tooltip we won't get any errors either way because we spread a `Record` type and in case of the error that will cause in this line: we can either ignore it or cast the type

```
describedBy: (rest as Record<string, string>['aria-describedby'],
```

Reply:
Can't think of any scenario where people would try to use this prop. We anyway have jsdoc with warning so should be fine. Having it in type will help us track which props are expected to be on dom.

Reply:
but if you out `ariaDescribedBy` via our makeA11yprops it will render `aria-describedby` only right?

Reply:
Yup. But in tooltip, we're passing `aria-describedby` directly (we have to because otherwise we'll have to put the checks saying if component is `Button` or `Link` use describedBy else use `aria-describedby` in custom components.